github/commitstatus.go

```diff
@@ -105,17 +106,18 @@ func (s CommitStatus) Add(sha, state, targetURL, description string) error {
 	var response httpResponse
 	timeToSleep := 0 * time.Second // 0 seconds
 
-	for attempt := 1; attempt <= maxRetries; attempt++ {
+	for attempt := 1; attempt <= s.maxRetries; attempt++ {
 		time.Sleep(timeToSleep)
-		s.log.Info("GitHub HTTP request", "attempt", attempt, "max", maxRetries)
+		s.log.Info("GitHub HTTP request", "attempt", attempt, "max", s.maxRetries)
 		response, err = httpRequestDo(req)
 		if err != nil {
 			return err
 		}
-		timeToSleep = checkForRetry(response)
+		timeToSleep = checkForRetry(response, s.waitTime, s.maxSleepTime)
 		if timeToSleep == 0 {
 			break
 		}
+		s.log.Info("Sleeping for:", timeToSleep)
```

Comment:
This will crash or report an error.
As I wrote in the previous comment, this is a structured logger and wants to be called as follows
```go
Info("this is the main message")
Info("this is the main message", "key1", value1, "key2", value2, ...)

```